src/Microsoft.Framework.Runtime.Roslyn/RoslynCompiler.cs

```diff
@@ -192,17 +193,21 @@ public class RoslynCompiler
 
         private void ApplyStrongNameSettings(CompilationContext compilationContext)
         {
-            // This is temporary, eventually we'll want a project.json feature for this
-            var keyFile = Environment.GetEnvironmentVariable(EnvironmentNames.BuildKeyFile);
+            var keyFile =
+                Environment.GetEnvironmentVariable(EnvironmentNames.BuildKeyFile) ??
+                compilationContext.Compilation.Options.CryptoKeyFile;
+
             if(!string.IsNullOrEmpty(keyFile))
             {
 #if DNX451
                 var delaySignString = Environment.GetEnvironmentVariable(EnvironmentNames.BuildDelaySign);
-                var delaySign = !string.IsNullOrEmpty(delaySignString) && (
-                    string.Equals(delaySignString, "true", StringComparison.OrdinalIgnoreCase) ||
-                    string.Equals(delaySignString, "1", StringComparison.OrdinalIgnoreCase));
+                var delaySign =
+                    delaySignString == null
+                        ? compilationContext.Compilation.Options.DelaySign
+                        : string.Equals(delaySignString, "true", StringComparison.OrdinalIgnoreCase) ||
+                          string.Equals(delaySignString, "1", StringComparison.OrdinalIgnoreCase);
 
-                var strongNameProvider = new DesktopStrongNameProvider();
+                var strongNameProvider = new DesktopStrongNameProvider(ImmutableArray.Create(Directory.GetCurrentDirectory()));
```

Comment:
do you mean we should not throw the error if delay sign is true and the file key is not provided and just skip the code? Why would it be useful?